src/rules/deprecation.ts

```diff
@@ -30,6 +30,37 @@ export type MessageIds = 'deprecated';
 
 type RequiredParserServices = ReturnType<typeof ESLintUtils.getParserServices>;
 
+const ruleCreator = (
```

Comment:
Is this suitable for what you suggested [here](https://github.com/gund/eslint-plugin-deprecation/pull/15#discussion_r527726936)?

Reply:
Oh no, it just got more complicated and still fails haha 🙈

Reply:
It's no problem, now the function is the same as it was before and at least it's not worse and we are not doing refactoring here so no worries =)

One more thing, please make it a proper function declaration instead of constant with arrow function, rename it to `createRuleForIdentifier` and move it below the `createRule` export.